src/qt/ve_qitems_mqtt.cpp

```diff
@@ -183,14 +183,21 @@ void VeQItemMqttProducer::open(
 		mWebSocket->setProtocol(
 			  protocolVersion == QMqttClient::MQTT_3_1 ? "mqttv3.1" : "mqtt");
 
+		connect(mWebSocket, &WebSocketDevice::disconnected,
+			this, [this] {
+				onStateChanged(QMqttClient::Disconnected);
+			}, Qt::QueuedConnection);
+
 		connect(mWebSocket, &WebSocketDevice::connected,
 			this, [this, protocolVersion] {
 				mMqttConnection->setProtocolVersion(protocolVersion);
 				mMqttConnection->setTransport(mWebSocket, QMqttClient::IODevice);
 				QMetaObject::invokeMethod(this, [this] { Q_EMIT aboutToConnect(); }, Qt::QueuedConnection);
-			});
+			}, Qt::QueuedConnection);
 
-		setConnectionState(Connecting);
+		if (connectionState() != Reconnecting) {
```

Comment:
I would prefer it if either `connectionState()` or `mConnectionState` was used consistently, rather than a mixture.

Reply:
I agree, connectionState() should be used throughout.  But I don't want to touch unrelated lines in this PR.